frontend/src/components/PieChart.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <div class="pieChart" 
+  style="
+  background-color:white;
+
+  ">
+      <Pie
+        :chart-options="chartOptions"
+        :chart-data="chartData"
+         />
+    </div>
+</template>
+
+<script>
+import { Pie } from 'vue-chartjs/legacy'
+import ChartDataLabels from 'chartjs-plugin-datalabels';
+import { Chart as ChartJS, Title, Tooltip, Legend, BarElement, CategoryScale,  ArcElement,
+ LinearScale } from 'chart.js'
+
+ChartJS.register(ChartDataLabels);
+ChartJS.register(Title, Tooltip, Legend, BarElement, CategoryScale, ArcElement,LinearScale)
+
+export default {
+  name: 'PieChart',
+  components: { Pie },
+  props: {
+    pieChartData: {
+      type: Object,
+      Default: {     
+       }
+    },
+  },
+  data() {
+
+    return {
+      chartData: {
+        labels: this.pieChartData["labels"],
+        datasets: [ {
+          backgroundColor:['#004c6d', '#7cd6ff', '#5e95b4', '#367090', '#86bdd9','#205e7e','#4b82a1','#72a9c6','#9ad1ec'],
+          data: this.pieChartData["UNIQUE_USER_COUNT"],       
+          datalabels: {
+            align: "start",
+            offset: -17,
+            anchor: function(context) {
+              let sum = 0;
+              for(let x in context.dataset.data){
+                sum = sum + context.dataset.data[x];
+              }
+              var index = context.dataIndex;
```

Comment:
In general, try to avoid using `var`. They give trouble with scope sometimes. (For example you could access var declared in for loop outside of it, because it's scoped to the function)  So it's better to use `const `as a default, or `let` when you know that the variable is going to change.